features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/edit/RoomDetailsEditPresenter.kt

```diff
@@ -61,41 +64,56 @@ class RoomDetailsEditPresenter @Inject constructor(
         val cameraPermissionState = cameraPermissionPresenter.present()
         val roomSyncUpdateFlow = room.syncUpdateFlow.collectAsState()
 
-        // Since there is no way to obtain the new avatar uri after uploading a new avatar,
-        // just erase the local value when the room field has changed
-        var roomAvatarUri by rememberSaveable(room.avatarUrl) { mutableStateOf(room.avatarUrl?.toUri()) }
+        val roomAvatarUri = room.avatarUrl()?.toUri()
+        var roomAvatarUriEdited by rememberSaveable { mutableStateOf<Uri?>(null) }
+        LaunchedEffect(roomAvatarUri) {
+            // Every time the roomAvatar change (from sync), we can set the new avatar.
+            roomAvatarUriEdited = roomAvatarUri
+        }
 
-        var roomName by rememberSaveable { mutableStateOf(room.displayName.trim()) }
-        var roomTopic by rememberSaveable { mutableStateOf(room.topic?.trim()) }
+        val roomRawNameTrimmed = room.rawName().orEmpty().trim()
+        var roomRawNameEdited by rememberSaveable { mutableStateOf("") }
+        LaunchedEffect(roomRawNameTrimmed) {
+            // Every time the rawName change (from sync), we can set the new name.
+            roomRawNameEdited = roomRawNameTrimmed
+        }
+        val roomTopicTrimmed = room.topic().orEmpty().trim()
+        var roomTopicEdited by rememberSaveable { mutableStateOf("") }
+        LaunchedEffect(roomTopicTrimmed) {
+            // Every time the topic change (from sync), we can set the new topic.
+            roomTopicEdited = roomTopicTrimmed
+        }
 
         val saveButtonEnabled by remember(
-            roomSyncUpdateFlow.value,
-            roomName,
-            roomTopic,
+            roomRawNameTrimmed,
```

Comment:
This is a bit weird to use `remember` with keys used in `derivatedStateOf`

Reply:
Yes, I tried to remove the keys, but this does not work as expected, for instance `roomRawNameTrimmed` is always empty.

Reply:
OK, so I have a mix in https://github.com/element-hq/element-x-android/pull/2849/commits/c3caff0a1f173082df7d72d77c03c9bb1acfb693. I think it's still relevant to use de `derivedStateOf` in this case.